queries/perl/highlights.scm

```diff
@@ -1,198 +1,107 @@
+(pod) @text
 
-; Built-in functions
-((call_expression
-  function_name: (identifier) @function.builtin)
- (#any-of? @function.builtin
-   "print" "printf" "sprintf" "say"
-   "push" "pop" "shift" "unshift" "splice"
-   "exists" "delete" "keys" "values"
-   "each"))
+(number) @number
+(version) @number
 
-[
-  (map)
-  (grep)
-  (bless)
-] @function.builtin
+(string_literal) @string
+(interpolated_string_literal) @string
+(quoted_word_list) @string
+(command_string) @string
 
-;; ----------
+[(heredoc_token) (command_heredoc_token)] @label
+(heredoc_content) @string
+(heredoc_end) @label
 
-(use_constant_statement
-  constant: (identifier) @constant)
+[(escape_sequence) (escaped_delimiter)] @string.escape
 
-(named_block_statement
-  function_name: (identifier) @function)
+[(quoted_regexp) (match_regexp)] @string.regex
 
-(function_definition
-  name: (identifier) @function)
+(autoquoted_bareword _?) @string.special
 
-(function) @function
+(hash_element_expression key: (bareword) @string.special)
 
-[
-  "(" ")"
-  "[" "]"
-  "{" "}"
-  (standard_input_to_variable)
-] @punctuation.bracket
+(use_statement (package) @type)
+(package_statement (package) @type)
+(require_expression (bareword) @type)
 
-[ "`" "\"" ] @punctuation.special
+(subroutine_declaration_statement name: (_) @function)
+(attribute_name) @attribute
+(attribute_value) @string
 
-[
-  "=~"
-  "!~"
-  "="
-  "=="
-  "+"
-  "-"
-  "."
-  "//"
-  "||"
-  "&&"
-  "<<"
-  (arrow_operator)
-  (hash_arrow_operator)
-  (array_dereference)
-  (hash_dereference)
-  (to_reference)
-  (type_glob)
-  (hash_access_variable)
-] @operator
+(goto_expression (label) @label)
+(loopex_expression (label) @label)
 
-[
-  (regex_option)
-  (regex_option_for_substitution)
-  (regex_option_for_transliteration)
-] @parameter
+(statement_label label: _ @label)
 
-(type_glob
-  (identifier) @variable)
+(relational_expression operator: "isa" right: (bareword) @type)
 
-[
-  (word_list_qw)
-  (command_qx_quoted)
-  (string_single_quoted)
-  (string_double_quoted)
-  (string_qq_quoted)
-  (bareword)
-  (transliteration_tr_or_y)
-] @string
+(function_call_expression (function) @function.call)
+(method_call_expression (method) @method.call)
+(method_call_expression invocant: (bareword) @type)
 
-[
- (heredoc_start_identifier)
- (heredoc_end_identifier)
-] @label
+(func0op_call_expression function: _ @function.builtin)
+(func1op_call_expression function: _ @function.builtin)
 
-(heredoc_body_statement) @text.literal
+(function) @function
 
-[
-  (pattern_matcher)
-  (regex_pattern_qr)
-  (patter_matcher_m)
-  (substitution_pattern_s)
-] @string.regex
+(ERROR) @error
 
-(escape_sequence) @string.escape
+; these are the ones nvim doesn't like
+[(scalar) (arraylen)] @variable.scalar
+(scalar_deref_expression ["->" "$" "*"] @variable.scalar)
+(array) @variable.array
+(array_deref_expression ["->" "@" "*"] @variable.array)
+(hash) @variable.hash
+(hash_deref_expression ["->" "%" "*"] @variable.hash)
```

Comment:
The "->" should be removed from here so it's highlighted as punctuation.
The rest may or may not be kept as is.

Reply:
I think i'll make it do both `@variable` and also `@variable.$type`, or is that not necessary?

Reply:
It will fall back to `@variable` anyway.